components/layout_2020/sizing.rs

```diff
@@ -96,3 +93,20 @@ pub(crate) fn outer_inline_content_sizes_and_percentages(
 
     (outer, pbm_percentages)
 }
+
+/// https://drafts.csswg.org/css2/visudet.html#shrink-to-fit-float
+pub(crate) fn shrink_to_fit(
+    content_sizes: &Option<ContentSizes>,
+    available_size: Length,
+) -> Length {
+    let content_sizes = expect(content_sizes);
+    available_size
+        .max(content_sizes.min_content)
+        .min(content_sizes.max_content)
+}
+
+fn expect(content_sizes: &Option<ContentSizes>) -> &ContentSizes {
+    content_sizes
+        .as_ref()
+        .expect("Accessing content size that was not requested")
+}
```

Comment:
What do you think of a structure with these two as a method and a single `Option<ContentSizes>` field?

Reply:
Added new enum.